tests/Microsoft.Identity.Test.Unit/WwwAuthenticateParametersTests.cs

```diff
@@ -56,6 +59,129 @@ public void CreateWwwAuthenticateResponse(string resource, string authorizationU
             Assert.IsNull(authParams.Error);
         }
 
+        [TestMethod]
+        [DataRow("AuthScheme", "")]
+        [DataRow("AuthScheme", "token68")]
+        [DataRow("AuthScheme", "auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        [DataRow("AuthScheme", "token68 auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        public void EnsureProperlyFormattedHeadersWithToken86DoNotFail(string scheme, string values)
+        {
+            // Arrange
+            HttpResponseMessage httpResponse = new HttpResponseMessage(HttpStatusCode.Unauthorized);
+            httpResponse.Headers.Add(WwwAuthenticateHeaderName, $"{scheme} {values}");
+
+            // Act
+            var authParams = WwwAuthenticateParameters.CreateFromAuthenticationHeaders(httpResponse.Headers, scheme);
+
+            // Assert
+            Assert.AreEqual(scheme, authParams.AuthScheme);
+
+            if (string.IsNullOrEmpty(values))
+            {
+                Assert.AreEqual(0, authParams.RawParameters.Count);
+            }
+            else if (values == "token68")
+            {
+                Assert.AreEqual("token68", authParams.RawParameters[scheme]);
+            }
+            else
+            {
+                Assert.AreEqual("token1", authParams.RawParameters["auth-param1"]);
+                Assert.AreEqual("token2", authParams.RawParameters["auth-param2"]);
+                Assert.AreEqual("token3", authParams.RawParameters["auth-param3"]);
+            }
+        }
+
+        [TestMethod]

```

Comment:
Does the current implementation support these scenarios?
- [a response can have multiple challenges with the same auth-scheme but with different realms.](https://httpwg.org/specs/rfc9110.html#rfc.section.11.5)
```csharp
httpResponse.Headers.Add(WwwAuthenticateHeaderName, "Bearer realm=\"first\", Bearer realm=\"second\"");
```
```csharp
httpResponse.Headers.Add(WwwAuthenticateHeaderName, "Bearer realm=\"first\"");
httpResponse.Headers.Add(WwwAuthenticateHeaderName, "Bearer realm=\"second\"");
 ```
- Header field [might contain more than one challenge](https://httpwg.org/specs/rfc9110.html#rfc.section.11.6.1)
```csharp
httpResponse.Headers.Add(WwwAuthenticateHeaderName, "Bearer realm=\"someRealm\", OtherScheme realm=\"someRealm\"");
```
```csharp
httpResponse.Headers.Add(WwwAuthenticateHeaderName, "Bearer realm=\"someRealm\"");
httpResponse.Headers.Add(WwwAuthenticateHeaderName, "OtherScheme realm=\"someRealm\"");
 ```